### ISSUE TITLE
NVIDIA Runtime Configuration Warning in Docker

### ISSUE THREAD

**Issue by fatemeh391** [NVIDIA/nvidia-container-toolkit#631]
I am experiencing a warning related to the NVIDIA runtime configuration while setting up Docker. Despite configuring the NVIDIA runtime correctly, I am seeing the following warning message:

sudo nvidia-ctk runtime configure --runtime=docker

WARN[0000] Ignoring runtime-config-override flag for docker
INFO[0000] Loading config from /etc/docker/daemon.json
INFO[0000] Wrote updated config to /etc/docker/daemon.json
INFO[0000] It is recommended that docker daemon be restarted.

Could you please help me understand why the runtime-config-override flag is being ignored and how to ensure that the NVIDIA runtime is properly set up as the default or specified runtime in Docker?

Thank you!

Fatemeh

**Comment by hosseybposh** [comment]
I have the same problem.

**Comment by RitchieMatt** [comment]
Same problem.  Spent most of the day trying to get around it.

**Comment by RitchieMatt** [comment]
**Finally fixed the issue!**


## Issue
Unable to change Docker runtime to `nvidia-container-runtime` despite proper installation of NVIDIA drivers and container toolkit.

## Solution
The key fix was removing a conflicting systemd override file:

```bash
sudo rm /etc/systemd/system/docker.service.d/override.conf
```

This override file was likely conflicting with the settings in `/etc/docker/daemon.json`.

## Explanation
1. The `daemon.json` file was correctly configured to use the NVIDIA runtime.
2. However, the systemd override file (`override.conf`) was overriding these settings, causing Docker to ignore the NVIDIA runtime configuration.
3. Removing the override file allowed Docker to use the settings from `daemon.json`.

## Lesson
A common issue in system administration: conflicting configurations across different levels (systemd overrides vs. application-specific config files).

### Best Practices
1. Always check for systemd overrides when troubleshooting service issues:
   ```bash
   systemctl cat docker.service
   ```
   This command shows the main unit file and any overrides.

2. Be cautious when creating systemd overrides. They can be powerful but can lead to unexpected behavior if not managed carefully.

3. When configuring Docker to use the NVIDIA runtime, prefer using the `daemon.json` file instead of systemd overrides.

4. If you need to use both, ensure they are not conflicting.

5. After making changes to Docker configuration, always restart the service and check its status:
   ```bash
   sudo systemctl restart docker
   sudo systemctl status docker
   ```

6. Verify the current runtime configuration:
   ```bash
   docker info | grep -i runtime
   ```

**Comment by elezar** [comment]
The `WARN[0000] Ignoring runtime-config-override flag for docker` warning can be ignored when configuring docker. This has to do with an incorrect default value in our CLI. 

@RitchieMatt there is definitely no connection between this warning and the `/etc/systemd/system/docker.service.d/override.conf` file. 

It may be that you're having other issues due to this.

**Comment by elezar** [comment]
I have created https://github.com/NVIDIA/nvidia-container-toolkit/pull/658 to address the warning.

**Comment by elezar** [comment]
This should be addressed in the v1.16.2 release. Please update and reopen if this is not the case.
